crypto/keys/keyring.go

```diff
@@ -50,31 +50,24 @@ func newKeyringKeybase(db keyring.Keyring, opts ...KeybaseOption) Keybase {
 
 // NewKeyring creates a new instance of a keyring. Keybase
 // options can be applied when generating this new Keybase.
+// Available backends are "os", "file", "test".
 func NewKeyring(
-	name string, dir string, userInput io.Reader, opts ...KeybaseOption,
+	svcName, backend, rootDir string, userInput io.Reader, opts ...KeybaseOption,
 ) (Keybase, error) {
-	db, err := keyring.Open(lkbToKeyringConfig(name, dir, userInput, false))
-	if err != nil {
-		return nil, err
-	}
 
-	return newKeyringKeybase(db, opts...), nil
-}
+	var db keyring.Keyring
+	var err error
 
-// NewKeyringFile creates a new instance of an encrypted file-backed keyring.
-func NewKeyringFile(name string, dir string, userInput io.Reader, opts ...KeybaseOption) (Keybase, error) {
-	db, err := keyring.Open(newFileBackendKeyringConfig(name, dir, userInput))
-	if err != nil {
-		return nil, err
+	switch backend {
+	case "test":
```

Comment:
IMHO these should be constants instead of the caller manually providing, say `"test"`.

Reply:
That was the original plan, but afterwards I forgot to refactor the strings into consts. Done now, thanks!